client/src/Pages/CreateActivity.jsx

```diff
@@ -0,0 +1,242 @@
+        };
+
+        const getUser = () => {
+            if (email !== "") {
+                try {
+                    if (data.data) {
+                        setUserId(data.data.find(user => user.email === email)._id);
+                    }
+                } catch (error) {
+                    console.error('getUser error:', error);
+                }
+            }
+        }
+
+        const getItinerary = async () => {
+            if (userId !== "") {
+                try {
+                    const res = await axios.get(`http://localhost:4000/itineraries/${userId}`, { withCredentials: true });
+                    const selectedItinerary = res.data.data.find(itinerary => itinerary._id === id);
+                    console.log(selectedItinerary);
+                    if (selectedItinerary) {
+                        setItineraryId(selectedItinerary._id);
+                        const startDate = dayjs(new Date(selectedItinerary.startDate)).format("MM/DD/YYYY");
+                        const endDate = dayjs(new Date(selectedItinerary.endDate)).format("MM/DD/YYYY");
+                        setItineraryStartDate(startDate);
+                        setItineraryEndDate(endDate);
+                        console.log(startDate, endDate);
+                    } else {
+                        console.error('Selected itinerary not found');
+                    }
+                } catch (error) {
+                    console.error('getItinerary error:', error);
+                }
+            }
+        }
+          
+        verifyCookie();
+        getUser();
+        getItinerary();
+    }, [navigate, cookies, removeCookie, email, data, id, userId]);
+
+    const handleError = (err) => {
+        toast.error(err, {
+            position: "bottom-left",
+        });
+    }
+
+    const handleSuccess = (msg) => {
+        toast.success(msg, {
+            position: "bottom-left",
+            autoClose: 3000,
+        });
+    }
+
+    const isValidUrl = urlString => {
+        var urlPattern = new RegExp('^(https?:\\/\\/)?'+ // validate protocol
+        '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // validate domain name
+        '((\\d{1,3}\\.){3}\\d{1,3}))'+ // validate OR ip (v4) address
+        '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ // validate port and path
+        '(\\?[;&a-z\\d%_.~+=-]*)?'+ // validate query string
+        '(\\#[-a-z\\d_]*)?$','i'); // validate fragment locator
+
+        return !!urlPattern.test(urlString);
+    }
+    console.log(isValidUrl(link));
+
+
+    const handleSave = async () => {
+        const data = {
+            title,
+            link,
+            address,
+            openingHours,
+            remark,
+            date
+        };
+        setLoading(true);
+
+        try {
+            if (title === "") {
+                setLoading(false);
+                handleError('Please fill out all fields');
+                return;
+            }
+
+            if (!isValidUrl(link)) {
```

Comment:
I think thre is an uncaught error (based on the activity model on the server side, the link provided needs to be unique hence when the link already exists in the database, the activity cannot be added). I think we can just change the activity model such that nothing has to be unqiue